src/level_four/core/controller/device_authentication_controller.c

```diff
@@ -130,259 +110,220 @@ static void helper_get_gendig_hash(atecc_slot_define_t slot,
-          continue;
-        }
-
-        atecc_data.status = atcab_sign_internal(
-            slot_2_auth_key, false, false, auth_serial_packet.signature);
-        if (atecc_data.status != ATCA_SUCCESS) {
-          continue;
-        }
-
-        atecc_data.status = atcab_read_zone(
-            ATCA_ZONE_DATA, slot_8_serial, 0, 0, auth_serial_packet.serial, 32);
-        if (atecc_data.status != ATCA_SUCCESS) {
-          continue;
-        }
-
-        helper_get_gendig_hash(slot_8_serial,
-                               auth_serial_packet.serial,
-                               tempkey_hash,
-                               auth_serial_packet.postfix1);
-
-        sign_internal_param.message = tempkey_hash;
-        sign_internal_param.digest = final_hash;
-
-        helper_sign_internal_msg(&sign_internal_param,
-                                 SIGN_MODE_INTERNAL,
-                                 slot_2_auth_key,
-                                 slot_8_serial);
-        {
-          uint8_t result = ecdsa_verify_digest(&nist256p1,
-                                               get_auth_public_key(),
-                                               auth_serial_packet.signature,
-                                               sign_internal_param.digest);
-          if (atecc_data.status != ATCA_SUCCESS || result != 0) {
-            LOG_ERROR(
-                "err xxx32 fault %d verify %d", atecc_data.status, result);
-            continue;
-          }
-        }
-      } while (--atecc_data.retries && atecc_data.status != ATCA_SUCCESS);
-      if (usb_irq_enable_on_entry == true)
-        NVIC_EnableIRQ(OTG_FS_IRQn);
-
-      memcpy(auth_serial_packet.postfix2,
-             &tempkey_hash[32],
-             POSTFIX2_SIZE);    // postfix 2 (12bytes)
-      transmit_data_to_app(DEVICE_SERAIL_NO_SIGNED,
-                           (uint8_t *)&auth_serial_packet,
-                           AUTH_DATA_SERIAL_SIGN_MSG_SIZE);
-
-#if X1WALLET_INITIAL
-      flow_level.level_three = DEVICE_AUTH_INFINITE_WAIT;
-      lv_task_set_prio(listener_task,
-                       LV_TASK_PRIO_MID);    // explicitly enable task listener
-    } break;
-
-    case DEVICE_AUTH_INFINITE_WAIT: {
-      // do nothing, just wait
-#endif
-    } break;
-
-    case SIGN_CHALLENGE: {
-      uint8_t nonce[32] = {0};
-      auth_data_t auth_challenge_packet = {0};
-      uint8_t io_protection_key[32] = {0};
-      uint8_t tempkey_hash[DEVICE_SERIAL_SIZE + POSTFIX2_SIZE] = {0};
-      uint8_t final_hash[32] = {0};
-      atca_sign_internal_in_out_t sign_internal_param = {0};
-      get_io_protection_key(io_protection_key);
+manager_auth_device_response_t __attribute__((optimize("O0")))
+sign_serial_number(void) {
+  manager_auth_device_response_t response =
+      MANAGER_AUTH_DEVICE_RESPONSE_INIT_ZERO;
+  response.which_response = MANAGER_AUTH_DEVICE_RESPONSE_SERIAL_SIGNATURE_TAG;
+
+  uint8_t nonce[32] = {0};
+  // auth_data_t auth_serial_packet = {0};
```

Comment:
Remove comment

Reply:
In sign challenge funcition and controller main too.

Reply:
Removed